### DISCUSSION TITLE
Person Ids as Allow List

### DISCUSSION THREAD

**Discussion by janstadt** [damongolding/immich-kiosk#25] [Ideas]
I think it would be nice to have the config.yaml for `person` be the allowed people to be viewed across all devices and then when a client wants to view a specific person, the query param value would need to be one of the allowed persons, otherwise default to the array of people in the persons. 

My use case is i have three kiosks i want to setup (one for each of my kids) and i want to be able to show the specific kid on a specific device and nothing else. (None of my kids want to see pics of their siblings. lol) 

Almost like an allow list and then the query params are per device/client but the ids have to be included in the persons in the config. (aka, i dont want any random crap from my college days somehow showing up on my kids devices when they figure out they can throw in a guid).

**Comment by damongolding** [comment]
What ever happened to sibling love?! /s

I think it's a interesting solution to your problem but I do see a bigger issue. If any of your kids were able to access and change the url they could just remove all the queries and have "access" to all your images.

A solution could be to set the base `immich_api_key` and `immich_url` values to "dummy" and give each child their own account, then share/add photos to those accounts. I would need to enable setting `immich_api_key` and `immich_url` via queries to be able to switch Immich instances though.

Bit of a tricky one!

**Comment by janstadt** [comment]
Thanks for the reply. If the kids were to modify the query params, they'd only be able to see what albums or people that are in the config file which would act like an allowlist. I do have our oldest on immich already as she can upload her own photos, but the two little ones who dont have phones wont have their own photos but yeah, i suppose i could share them specific albums that only include their photos or photos of them and their friends or something. Certainly puts the onus on immich for managing what each kiosk can see vs the kiosk docker container though.

**Comment by damongolding** [comment]
The more I think about it. The more it makes sense to have the people list work as a white list. I’m planning to implement multiple albums, just like how people work at the moment. Once that’s implemented I would then be able to use them as whitelist/lock them.

**Reply by Kernald** [reply]
I was about to open a similar-ish feature request: being able to exclude specific persons. There are photos of some people in my library that I'd like to keep but not to see everyday.
